h2/src/main/org/h2/command/dml/Select.java

```diff
@@ -847,9 +854,11 @@ public void prepare() {
             sort = prepareOrder(orderList, expressions.size());
             orderList = null;
         }
-        for (int i = 0; i < expressions.size(); i++) {
-            Expression e = expressions.get(i);
-            expressions.set(i, e.optimize(session));
+        if (!isExistsSubquery) {
+            for (int i = 0; i < expressions.size(); i++) {
+                Expression e = expressions.get(i);
+                expressions.set(i, e.optimize(session));
+            }
```

Comment:
I admit, I'm was a bit nervous about this, myself, but then, what can possibly go wrong? The `Expression.optimize(Session)` call should return an equivalent expression to the original one, so it should be entirely optional. Given that it is avoidable extra work, what regression risk do we have (apart from skipping code that shouldn't be in optimize in the first place)?

The reason for this implementation is that the optimisation step would evaluate `1 / 0`, being a constant expression, prior to execution, which shouldn't be done in this case. If there was a way to delay this optimisation step...?